### DISCUSSION TITLE
Harbor upgrade from 2.6.3 to 2.8.0 and breaking changes

### DISCUSSION THREAD

**Discussion by slacki123** [goharbor/harbor#18543] [Q&A] [answered]
Hi everyone, I hope this is the correct place to ask.

I am going to soon be looking to upgrade our production harbor with existing data from 2.6.3 to 2.8.0, however it's taking me too long to understand how safe this is.

The other safe option is to do a blue/green deployment, migrate all data to the new harbor and start using that, but obviously that's a lot more effort.

Ideally I'd like to upgrade harbor in place by just changing the helm chart version, and redeploying everything with incremented versions, but I am not confident about that. Would I need to redeploy any databases, run any database migration scripts, change anything in my s3 buckets etc?

Is there anything in particular I should out for when upgrading? 
Alternatively how does one typically find information on these things?

Any insights would be appreciated

Thanks!

**Comment by Vad1mo** [comment, marked as answer]
make a backup of your database before upgrade, to be on the safe side. The S3 bucket is not touched during upgrades, so there is no need to make a backup. I would also recommend cleaning Redis after the upgrade.
